packages/offix-datastore/src/Model.ts

```diff
@@ -89,4 +133,34 @@ export class Model<T = unknown> {
                 listener(event);
             });
     }
+
+    private async subscribeForServerEvents(replicator: IReplicator) {
+        // TODO replicator.subscribe
+        // handle subscription events here
+    }
+
+    private async doDeltaSync(replicator: IReplicator, matcher: (d: T) => Predicate<T>, predicate?: Predicate<T>) {
+        // TODO limit the size of data returned
+        // TODO get lastSync for model for metadata store and pass to pullDelta
+        const data = await replicator.pullDelta(this.getStoreName(), "", predicate);
+
+        data
+            .filter((d: any) => (d._deleted))
```

Comment:
[Discussion] I think we should still save deleted item for conflict resolution - purposes. Without that we will need to force deltas for every modification that happens on the server as we can have local conflicts.